### DISCUSSION TITLE
ChatGPT and Dalle Integration

### DISCUSSION THREAD

**Discussion by SatireSage** [CSSS/wall_e#385] [Ideas]
The ChatGPT and Dalle integration is an advanced feature added to the Computing Science Student Society (CSSS) Discord bot, Wall-E, allowing users to have more engaging and intelligent conversations.

Users can now engage in more natural and engaging conversations with Wall-E (through ChatGPT) receive personalized assistance or academic support, and get the answers to their questions quickly and easily, without having to wait or rely on other users. Additionally through Dalee, the bot will be able to provide a more dynamic and visually exciting conversation.

I thought it would be great if CSSS could include it in the bot as well. If that's possible, please reach out to me since I would love to contribute! I have created a discord bot called EngBuddy that accomplishes this for the Engineering Science Student Society (ESSS) so please feel free to reach out to me if you wish to test it out.

<img width="1419" alt="Screenshot 2023-03-26 at 8 17 15 AM" src="https://user-images.githubusercontent.com/49692422/227817326-4568d3ba-e60b-4b72-ace4-e47b2e75dc80.png">
<img width="1454" alt="Screenshot 2023-03-26 at 8 19 38 AM" src="https://user-images.githubusercontent.com/49692422/227817313-95bd70d0-0c75-48cf-a19b-50678409f82a.png">

**Comment by modernNeo** [comment]
Hello,

If you want to add ChatGPT to the bot, I have nothing against it as long as it does not result in a negative bot experience. 

But you will first need to clear it with the moderators of the CSSS Discord guild. And I personally won't be of much help as I have no experience with that particular technology but I can help with any architecture questions/changes that may be needed.

**Reply by SatireSage** [reply]
Hey! 

Yes the integration doesn't result in a negative experience since the bot doesn't answer offensive or inappropriate questions. Should I reach out to any moderator in the server?

Also, since the Wall-E bot and my bot are both based on python I don't believe architecture changes would be required but yes I shall definitely get back to you after speaking to the moderators!

Thanks

**Comment by modernNeo** [comment]
> Yes the integration doesn't result in a negative experience since the bot doesn't answer offensive or inappropriate questions. 

I mean any lag that might result from using the ChatGPT code.

> Should I reach out to any moderator in the server?

Go to https://discord.sfucsss.org and ask on the [server-suggestions](https://discord.com/channels/228761314644852736/462373784897454101) channel

**Reply by SatireSage** [reply]
> I mean any lag that might result from using the ChatGPT code.

The only time the user might expect to wait for a second or two would be when making a ChatGPT or Dalle call and is handled by a message such as the following:
<img width="350" alt="Screenshot 2023-03-26 at 6 59 41 PM" src="https://user-images.githubusercontent.com/49692422/227823403-e69559d6-71a2-4e94-bdab-d1eadb95537d.png">
That being said I believe the rest of the bot's commands shall remain unaffected!

> Go to https://discord.sfucsss.org/ and ask on the [server-suggestions](https://discord.com/channels/228761314644852736/462373784897454101) channel

Sure thing! Thanks once again

**Comment by modernNeo** [comment]
an alternative route I have found that works is implementing the feature and then asking the Mods if they are against it. they don't always know what they want 😉
